src/lxc/conf.c

```diff
@@ -623,6 +623,37 @@ unsigned long add_required_remount_flags(const char *s, const char *d,
 #endif
 }
 
+static int add_shmount_to_conf(struct lxc_conf *conf, char *full_cont_path) {
+	char *new_mount;
+	size_t len_mount;
+	int ret = -1;
+
+	/* +1 for the separating whitespace */
+	len_mount = strlen(conf->lxc_shmount.path_host) + 1
+			+ strlen(full_cont_path) + sizeof(" none bind,create=dir 0 0") - 1;
+	/* +1 for the termintating '\0' */
+	new_mount = malloc(len_mount + 1);
+	if (!new_mount) {
+		SYSERROR("Not enough memory");
+		return -ENOMEM;
+	}
+
+	ret = snprintf(new_mount, len_mount + 1, "%s %s none bind,create=dir 0 0",
+				   conf->lxc_shmount.path_host, full_cont_path);
+	if (ret < 0 || ret >= len_mount + 1) {
+		ret = -1;
+		goto out_free_new_mount;
+	}
```

Comment:
In such cases, where there's only one jump location I would suggest to use:
```
if (/* error */) {
        free(new_mount);
        return -1;
}
```
but you can leave it as is. :)